apps/open-swe/src/graphs/programmer/nodes/take-action.ts

```diff
@@ -85,6 +91,40 @@ export async function takeAction(
     throw new Error("No tool calls found.");
   }
 
+  // Filter out unsafe commands
+  const { filteredToolCalls, wasFiltered } = await filterUnsafeCommands(
+    toolCalls,
+    config,
+  );
+
+  if (wasFiltered) {
+    // If all tool calls were filtered out, we need to handle this differently
+    if (filteredToolCalls.length === 0) {
+      // Remove the last message entirely since it has no valid tool calls
+      const modifiedMessages = state.internalMessages.slice(0, -1);
+      return new Command({
+        goto: "take-action",
+        update: { internalMessages: modifiedMessages },
+      });
+    }
+
+    // Create a modified message with only safe tool calls
+    const modifiedMessage = {
+      ...lastMessage,
+      tool_calls: filteredToolCalls,
+    };
```

Comment:
and recreate it like this:
```suggestion
    const modifiedMessage = new AIMessage({
      ...lastMessage,
      tool_calls: filteredToolCalls,
    });
```